cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -493,5 +496,27 @@ func main() {
 		}),
 		Conns:              connset,
 		RefreshCfgThrottle: refreshCfgThrottle,
-	}).Run(connSrc)
+	}
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGTERM)
```

Comment:
consider also listening for SIGINT, that way this will also work when run interactively and the user hits ctrl-c